esphome/components/xpt2046/__init__.py

```diff
@@ -0,0 +1,124 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome import pins
+from esphome.components import spi
+from esphome.const import CONF_ID, CONF_ON_STATE, CONF_THRESHOLD, CONF_TRIGGER_ID
+
+CODEOWNERS = ["@numo68"]
+AUTO_LOAD = ["spi", "binary_sensor"]
```

Comment:
This should be a dependency, not an auto-load because SPI requires configuration and cannot be auto loaded.
```suggestion
AUTO_LOAD = ["binary_sensor"]
DEPENDENCIES = ["spi"]
```